File: C++/oddEvenList.cpp
#include<iostream>

struct ListNode {
 	int val;
	struct ListNode* next;
	ListNode(int x) : val(x), next(nullptr) {}
};

class Solution{
    public:
     ListNode* oddEvenList(ListNode* head) {
        if(head==nullptr) {return nullptr;}
        ListNode *evenhead=head->next, *odd=head, *even=evenhead;
        while (even&&even->next) {
            odd->next = even->next;
            even->next = even->next->next;
            odd = odd->next;
            even = even->next;
        }
        odd->next = evenhead;
        return head;
    }
};

int main(){
    ListNode* head=new ListNode(1);
    head->next=new ListNode(2);
    head->next->next=new ListNode(3);
    head->next->next->next=new ListNode(4);
    head->next->next->next->next=new ListNode(5);
    Solution solution;

    for(ListNode* node=head;node !=nullptr;node=node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    ListNode* oddevenHead=solution.oddEvenList(head);
    for(ListNode* node=oddevenHead;node !=nullptr;node=node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    while(head){
        ListNode* next=head->next;
        delete head;
        head=next;
    }

    return 0; 
}

File: python/oddEvenList.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def oddEvenList(self , head: ListNode) -> ListNode:
        if head is None:
            return head
        evenhead =head.next
        odd = head
        even =evenhead
        while even and even.next:
            odd.next = even.next
            even.next = even.next.next
            odd = odd.next
            even = even.next
        odd.next = evenhead
        return head

head=ListNode(1)
head.next=ListNode(4)
head.next.next=ListNode(6)
head.next.next.next=ListNode(3)
head.next.next.next.next=ListNode(7)
solution=Solution()
odd_evenhead=solution.oddEvenList(head)
while odd_evenhead:
    print(odd_evenhead.val,end='')
    odd_evenhead=odd_evenhead.next
print()